src/main/java/org/ice4j/ice/Component.java

```diff
@@ -920,10 +920,38 @@ private void free(LocalCandidate localCandidate)
      */
     public LocalCandidate findLocalCandidate(TransportAddress localAddress)
     {
+        return findLocalCandidate(localAddress, null);
+    }
+
+    /**
+     * Returns the local <tt>LocalCandidate</tt> with the specified
+     * <tt>localAddress</tt> if it belongs to this component or <tt>null</tt>
+     * if it doesn't. If {@code base} is also specified, looks for a candidate
+     * whose base matches {@code base}.
+     *
+     * @param localAddress the {@link TransportAddress} we are looking for.
+     * @param base an optional base to match.
+     *
+     * @return  the local <tt>LocalCandidate</tt> with the specified
+     * <tt>localAddress</tt> if it belongs to this component or <tt>null</tt>
+     * if it doesn't.
+     */
+    public LocalCandidate findLocalCandidate(TransportAddress localAddress, LocalCandidate base)
+    {
+        for (LocalCandidate localCnd : localCandidates)
+        {
+            if (localCnd.getTransportAddress().equals(localAddress))
```

Comment:
This was intentional to preserve the previous behavior because I couldn't convince myself it doesn't break anything. I've added a comment to make it clear.